e2etest/pkg/metrics/metrics.go

```diff
@@ -11,26 +11,39 @@ import (
 
 	dto "github.com/prometheus/client_model/go"
 	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
 	"go.universe.tf/metallb/e2etest/pkg/executor"
+
 	corev1 "k8s.io/api/core/v1"
 
 	"github.com/pkg/errors"
 )
 
+type PrometheusResponse struct {
+	Status string                 `json:"status"`
+	Data   prometheusResponseData `json:"data"`
+}
+
+type prometheusResponseData struct {
+	ResultType string       `json:"resultType"`
+	Result     model.Vector `json:"result"`
+}
+
 // MetricsForPod returns the parsed metrics for the given pod, scraping them
 // from the source pod.
 func ForPod(source, target *corev1.Pod, namespace string) ([]map[string]*dto.MetricFamily, error) {
 	ports := make([]int, 0)
 	allMetrics := make([]map[string]*dto.MetricFamily, 0)
 	for _, c := range target.Spec.Containers {
 		for _, p := range c.Ports {
-			if p.Name == "monitoring" {
+			if p.Name == "monitoring" ||
+				p.Name == "frr-monitoring" {
 				ports = append(ports, int(p.ContainerPort))
 			}
 		}
 	}
 
-	podExecutor := executor.ForPod(namespace, source.Name, source.Spec.Containers[0].Name)
```

Comment:
this means the source pod has to be the controller pod? shouldn't we take the container's name as a parameter instead?

Reply:
I'd just rename source -> controller, because the reality is, we always use the controller.
It's better than bloating the code with a parameter that is always the same